### PR TITLE
Add CD Pipeline and vscode launch task

### PR DESCRIPTION
**Pull request by machteldbogels**
This PR Includes:
* CD pipeline for deploying the terraform scripts and running the E2E test through GitHub Actions
* Adding terraform outputs necessary for running the CD pipeline
* Adding launch tasks for debugging in VS Code
* Changing the log severity to Warning for the Microsoft category within Application Insights

### CODE REVIEW

**Review by Ioana37 — APPROVED** [review]

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    defaults:
+      run:
+        working-directory: ./terraform
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Ensure directory permissions
+        run: sudo -s
```
Comment (syedhassaanahmed) [inline .github/workflows/cd.yml:23]: Why do we need this? On which specific directory and what kind of permissions are needed? If a permission error happens, which part of the Terraform template causes it?

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          # This wrapper must be disabled in order to correctly capture output from terraform CLI into a bash var
+          # See https://github.com/hashicorp/setup-terraform/issues/20
+          terraform_wrapper: false
+
+      - name: Terraform Init
+        run: terraform init
+
```
Comment (syedhassaanahmed) [inline .github/workflows/cd.yml:37]: Should we add a `terraform plan` step in between to validate the state and allow us to fail early?

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    defaults:
+      run:
+        working-directory: ./E2ETests
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Mask IoT Hub Connection String
+        run: echo "::add-mask::$IotHubConnectionString"
```
Comment (syedhassaanahmed) [inline .github/workflows/cd.yml:80]: Very cool! Didn't know about this syntax for masking secrets.

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+      - name: Enable public network access for IoT Hub
+        run: |
+          az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID
+          az extension add --name azure-iot
+          az iot hub update --name $IotHubName --set properties.publicNetworkAccess=Enabled
+
+      - name: Run E2E Test
+        run: dotnet test -e IotHubConnectionString=$IotHubConnectionString -e EdgeDeviceName=$EdgeDeviceName
+
```
Comment (syedhassaanahmed) [inline .github/workflows/cd.yml:90]: Should we enhance the CD pipeline to also display nicely formatted test results? We could try this [marketplace Action](https://github.com/marketplace/actions/test-reporter) which can interpret the [test results](https://github.com/Microsoft/vstest-docs/blob/main/docs/report.md) output from `dotnet test`

Comment (syedhassaanahmed) [reply .github/workflows/cd.yml:90]: As agreed in the chat with @Ioana37, this is nice to have. Let's revisit this once we have more tests.

File: .github/workflows/cd.yml (hunk 1 above)
Comment (machteldbogels) [reply .github/workflows/cd.yml:23]: I added it after I received the below errors during terraform apply on the /opt directory:
image

as well as this directory: 
image

I could look into adding specific directory permissions if that is preferred

File: .github/workflows/cd.yml (hunk 2 above)
Comment (machteldbogels) [reply .github/workflows/cd.yml:37]: Done! thanks

File: .github/workflows/cd.yml (hunk 1 above)
Comment (syedhassaanahmed) [reply .github/workflows/cd.yml:23]: ok, can we instead grant permission on the specific directory
`sudo chown -R $(id -u) /home/runner/.azure`
And add a comment about why this step is needed. This sounds like a bug to me in Azure CLI.

Comment (Ioana37) [reply .github/workflows/cd.yml:23]: I've tried the command you suggested, but that directory is not yet created when running this step.
dir permissions

Comment (syedhassaanahmed) [reply .github/workflows/cd.yml:23]: ok, then it definitely sounds like something AzCLI should handle. Let's add a note about it and revert back to the previous sudo

**Review by syedhassaanahmed — APPROVED** [review]
Solid work once again @machteldbogels and @Ioana37  🚀

File: .github/workflows/cd.yml (hunk 1 above)
Comment (machteldbogels) [reply .github/workflows/cd.yml:23]: This `$(id -u)` is nice to know though, I tried in an earlier [commit](https://github.com/machteldbogels/keyvault-ca/commit/d0701b789a11b04fc8a5cd72ab6458cdbdaab464) with guesstimating 'user' but that could have taken a long time to figure out the right one. 😊

**Pull request merged**